model_versions/reserved_versions.go

```diff
@@ -0,0 +1,13 @@
+package model_versions
+
+import "strings"
+
+const (
+	XAkitaLatestModelVersion = "latest"
+)
+
+// Determines whether a version is reserved for Akita internal use.
+func IsReservedModelVersion(k string) bool {
+	s := strings.ToLower(string(k))
+	return strings.EqualFold(s, XAkitaLatestModelVersion)
+}
```

Comment:
We also use the x-akita-big-model prefix for internal versions.

https://github.com/akitasoftware/superstar/blob/467ec50cac624b4c1b7a44d7615ef3d8a6e740e5/lib/versions/versions.go#L15